src/renderer/ipc/invalid-kubeconfig-handler.tsx

```diff
@@ -0,0 +1,44 @@
+import React from "react";
+import { ipcRenderer, IpcRendererEvent, shell } from "electron";
+import { clusterStore } from "../../common/cluster-store";
+import { InvalidKubeConfigArgs, InvalidKubeconfigChannel } from "../../common/ipc/invalid-kubeconfig";
+import { Notifications, notificationsStore } from "../components/notifications";
+import { Button } from "../components/button";
+
+export const invalidKubeconfigHandler = {
+  source: ipcRenderer,
+  channel: InvalidKubeconfigChannel,
+  listener: InvalidKubeconfigListener,
+  verifier: (args: [unknown]): args is InvalidKubeConfigArgs  => {
+    return args.length === 1 && typeof args[0] === "string" && !!clusterStore.getById(args[0]);
+  },
+};
+
+function InvalidKubeconfigListener(event: IpcRendererEvent, ...[clusterId]: InvalidKubeConfigArgs): void {
+  const notificationId = `invalid-kubeconfig:${clusterId}`;
+  const cluster = clusterStore.getById(clusterId);
+
+  Notifications.error(
+    (
+      <div className="flex column gaps">
+        <b>Cluster with invalid Kubeconfig Detected!</b>
+        <p>Cluster <b>{cluster.name}</b> has invalid kubeconfig and cannot be displayed.
+        Please fix the <a href="#" onClick={(e) => { e.preventDefault(); shell.showItemInFolder(cluster.kubeConfigPath); }}>kubeconfig</a> manually or remove the cluster. </p>
+        <p>Do you want to remove the cluster now?</p>
+        <div className="flex gaps row align-left box grow">
+          <Button active outlined label="Remove" onClick={()=> {
+            clusterStore.removeById(clusterId);
+            notificationsStore.remove(notificationId);
+          }} />
+          <Button active outlined label="Cancel" onClick={() => notificationsStore.remove(notificationId)} />
```

Comment:
Shouldn't the notification be removed here too?

Reply:
What do you mean? To remove the notification when clicking `Cancel` button? It should already do that.

Reply:
Looking at the code, there doesn't seem to be any code inspecting a notification's `message` field for a button labelled with `"Cancel"`. There also doesn't seem to be any other instances of this pattern too (which is why I didn't know about it if it is the case).

Reply:
The only place where a button has a label of `"Cancel"` is in `InfoPanel` but that is for the hover text

Reply:
Oh sorry, I completely blanked while reading the code. Sorry.

Reply:
`notificationsStore.remove(notificationId)` should remove the notification or at least it disappears when I clicked the `Cancel` button.

Reply:
Yes I don't know why I thought otherwise. Sorry.